src/components/VirtualTable/TableHead.tsx

```diff
@@ -95,34 +195,19 @@ export const TableHead = <T,>({
             <thead className={b('head')}>
                 <tr>
                     {columns.map((column) => {
-                        const content = column.header ?? column.name;
                         const sortOrder =
                             sortParams.columnId === column.name ? sortParams.sortOrder : undefined;
 
                         return (
-                            <th
+                            <TableHeadCell
                                 key={column.name}
-                                className={b(
-                                    'th',
-                                    {align: column.align, sortable: column.sortable},
-                                    column.className,
-                                )}
-                                style={{
-                                    height: `${rowHeight}px`,
-                                }}
-                                onClick={() => {
-                                    handleSort(column.name);
-                                }}
-                            >
-                                <div className={b('head-cell')}>
-                                    {content}
-                                    <ColumnSortIcon
-                                        sortOrder={sortOrder}
-                                        sortable={column.sortable}
-                                        defaultSortOrder={defaultSortOrder}
-                                    />
-                                </div>
-                            </th>
+                                column={column}
+                                sortOrder={sortOrder}
+                                defaultSortOrder={defaultSortOrder}
+                                onSort={handleSort}
+                                rowHeight={rowHeight}
+                                resizeObserver={resizeObserver.current}
```

Comment:
It seems it would be better not to pass `resizeObserver`, but make a component with `forwardRef` and do all observe staff here.

Reply:
I intentionally created `TableHeadCell` component to `observe` and `unobserve` `<div>` on mount and unmount in `useEffect`. The reason - it's very important to correctly `unobserve` `<div>` on unmount, otherwise some odd values could be stored as column width.

In case with `forwardRef`, it's needed to store some reference to initial divs, to manually `unobserve` them on their unmount. So with removing `useEffect` construction from `HeadCell` some more complex construction will appear in `TableHead`

Reply:
Nonetheless, it's not correct to pass `ref.current` as property, because React doesn't guarantee to rerender children if `ref.current` will change.
So, as for me, array with refs is better solution. You may pass `unsubscribe` method to all `th`s.

Reply:
Always forget about this issue with `useRef`.

Replaced it with `useState`, now it should rerender properly.

As for the array of refs option, it won't be just array of refs, because there are cases, where columns could be unmounted one by one. Examples: tables with conditional columns - `Storage`, where degraded columns is shown only with specific filters setup, `Partitions`, where some columns appear, if consumer is selected; tables with columns setup - you can hide and show specific columns. So it either would be some additional events handlers, or some logic, to process this array to subscribe new columns and unsubscribe disappeared.

Reply:
What if we create functions in `TableHead` and pass it to all cells? Or this logic may be moved to custom hook.

```
const observedCells = React.useRef({})

const {observeCell, unobserveCell} = React.useMemo(() => {
  const resizeObserver = new ResizeObserver(......)

  const unobserveCell = (cellId) => {
  const cellRef = observedCells.current.cellId
  if (cellRef) {
    resizeObserver.unobserve(cellRef)
    delete observedCells.current.cellId
  }
}
const observeCell = (cellId, cellRef) => {
  unobserveCell(cellId)
  resizeObserver.unobserve(cellRef)
  observedCells.current.cellId = cellRef
}

  return {observeCell, unobserveCell}

}, [onColumnsResize, isTableResizeable])
```